amethyst_network/src/simulation/resource.rs

```diff
@@ -0,0 +1,210 @@
+use super::{
+    client::Client,
+    message::Message,
+    requirements::{DeliveryRequirement, UrgencyRequirement},
+    transport::socket::Socket,
+};
+use std::{collections::VecDeque, net::SocketAddr};
+
+/// Resource serving as the owner of the underlying socket and the queue of messages to be sent.
+pub struct NetworkSimulationResource<S: Socket> {
+    socket: Option<S>,
+    is_server: bool,
```

Comment:
On a higher level design note of those with clients and servers and remote connections. As discussed in the other comment, there is the issue of knowing clients and active game/lobby joining. I'd also like the bring up the P2P scenario; currently this is pretty strictly designed to a client-server architecture. It may be worth making sure to keep in consideration a strictly authoritative based approach as well. This would be something akin:
- Any engine can create entities
- Engines can give away certain authorities to other engines (update, etc)
- Engines can take back authority on any entities they own.

This type of mechanism allows for distributing some entity updates and ownership either between multiple servers, or giving the clients entity authority over a single entity; allowing them to just update their 'player' entity.  

This would require stepping this away from teh design of a `client` and `server`, and instead to a `matchmaker` and `engine`. A matchmaker can also be an engine, but strictly handles providing dispatching to an `engine` for requesting synchronization and authority from.

Reply:
That is a fair point. I'm likely conflating levels of abstraction by baking this information in here. Presumably, we'd want a higher level to make the determination of which `SocketAddr`s to send messages to and have this resource strictly own the socket and queue messages for the underlying systems to churn through.